logger.ts

```diff
@@ -109,5 +113,39 @@ export class Logger implements ILogger {
 
 		return printMsgFuture;
 	}
+
+	private getPasswordEncodedArguments(args: string[]): string[] {
+		return _.map(args, a => {
+			var passwordRegex = /[Pp]assword=(.*?)(['&,]|$)|\"[Pp]assword\":\"(.*?)\"/,
```

Comment:
The regexp creation should happen only once. Consider moving it outside the map lambda.